test/test_log_forwarding.py

```diff
@@ -0,0 +1,134 @@
+import gzip
+from io import (
+    StringIO,
+)
+import json
+from unittest.mock import (
+    MagicMock,
+    patch,
+)
+
+import chalice.app
+from moto import (
+    mock_s3,
+)
+
+from app_test_case import (
+    LambdaEnvTestCase,
+)
+from azul import (
+    cached_property,
+    config,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.indexer.log_forwarding_controller import (
+    LogForwardingController,
+)
+from azul.service.storage_service import (
+    StorageService,
+)
+
+
+class TestLogForwarding(LambdaEnvTestCase):
+
+    @property
+    def log_bucket(self) -> str:
+        return aws.qualified_bucket_name(config.logs_term)
+
+    @property
+    def log_file_key(self) -> str:
+        prefix = config.alb_access_log_path_prefix('gitlab')
+        return '/'.join([prefix, 'AWSLogs', '123123123123', 'elasticloadbalancing', '2023', '01', '01', 'test.log.gz'])
+
+    @cached_property
+    def storage_service(self) -> StorageService:
+        return StorageService(bucket_name=self.log_bucket)
+
+    @mock_s3
+    def test(self):
+        self.storage_service.create_bucket()
+        event = chalice.app.S3Event(context={}, event_dict={
+            'Records': [{
+                's3': {
+                    'bucket': {'name': self.log_bucket},
+                    'object': {'key': self.log_file_key}
+                }
+            }]
+        })
+
+        # Test that troublesome characters appear verbatim in the output
+        for chars in (
+            '\\',
```

Comment:
If I'm right about the escaping, a single backslash cannot occur. It would be escaped as `\x22`.